# Обработка исключений

Обработка исключений - механизм обработки ошибок и необычных ситуаций в программе.
Когда в процессе выполнения кода возникает ошибка, Python создает объект исключения и пытается найти соответствующий блок обработки исключений.

```python
try:
    x = int(input("Введите число: "))
    result = 10 / x
except ValueError as e:
    print(f"Ошибка: Введено не число. {e}.")
except ZeroDivisionError:
    print("Ошибка: Деление на ноль.")
else:
    print(f"Результат: {result}")
finally:
    print("Завершение программы.")
```

**Основные элементы**

- try: В этом блоке помещается код, в котором может произойти исключение.
- except: Здесь указывается тип исключения, которое вы хотите перехватить. Если происходит исключение этого типа, управление передается в соответствующий блок except.
- as e: Переменная e используется для хранения информации об исключении, что позволяет получить дополнительные сведения о произошедшей ошибке.
- else: Этот блок выполняется, если в блоке try не произошло исключений.
- finally: Код в этом блоке выполняется всегда, независимо от того, произошло исключение или нет. Это место, где обычно размещают код для завершающих действий, таких как закрытие файлов или сетевых соединений.
